build-tools/Xamarin.Android.Tools.BootstrapTasks/Xamarin.Android.Tools.BootstrapTasks/CheckApiCompatibility.cs

```diff
@@ -0,0 +1,225 @@
+				return !Log.HasLoggedErrors;
+			}
+
+			// Get the previous api implementation path by replacing the current api string with the previous one.
+			var previousTargetImplementationPath = TargetImplementationPath.Replace (ApiLevel, previousApiLevel);
+
+			// In case previous api is not defined or directory does not exist we can skip the check.
+			var validateAgainstPreviousApi = !(string.IsNullOrWhiteSpace (previousApiLevel) || !Directory.Exists (previousTargetImplementationPath));
+			if (validateAgainstPreviousApi) {
+
+				// First we check the Api level assembly against the previous api level assembly
+				// i.e.: check api breakages using "the just built V2.dll" against "the just built V1.dll"
+				ValidateApiCompat (previousTargetImplementationPath);
+
+				if (Log.HasLoggedErrors) {
+					return !Log.HasLoggedErrors;
+				}
+			}
+
+			// If Api level is the latest we should also compare it against the reference assembly
+			// located on the external folder. (xamarin-android-api-compatibility)
+			// i.e.: check apicompat using "the just built V2.dll" against V2.dll located on xamarin-android-api-compatibility repo
+			if (ApiLevel == LastStableApiLevel) {
+
+				// Check xamarin-android-api-compatibility reference directory exists
+				var referenceContractPath = Path.Combine (ApiCompatContractPath, "reference");
+				if (!Directory.Exists (referenceContractPath)) {
+					Log.LogMessage (MessageImportance.High, $"CheckApiCompatibility Warning: Skipping reference contract check.\n{referenceContractPath} does not exist.");
+					return !Log.HasLoggedErrors;
+				}
+
+				ValidateApiCompat (referenceContractPath);
+			}
+
+			return !Log.HasLoggedErrors;
+		}
+
+		// Validates Api compatibility between contract (previous version) and implmentation (current version)
+		// We do that by using Microsoft.DotNet.ApiCompat.dll
+		void ValidateApiCompat (string contractPath) {
+
+			var apiCompat = Path.Combine (ApiCompatPath, "Microsoft.DotNet.ApiCompat.exe");
+
+			string content;
+			using (var genApiProcess = new Process()) {
+
+				if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows)){
+  					genApiProcess.StartInfo.FileName = apiCompat;
+					genApiProcess.StartInfo.Arguments = $"{contractPath} -i {TargetImplementationPath}";
+				} else {
+					genApiProcess.StartInfo.FileName = $"mono";
+					genApiProcess.StartInfo.Arguments = $"{apiCompat} {contractPath} -i {TargetImplementationPath}";
+				}
+
+				genApiProcess.StartInfo.UseShellExecute = false;
+				genApiProcess.StartInfo.CreateNoWindow = true;
+				genApiProcess.StartInfo.RedirectStandardOutput = true;
+
+				// Get api definition for previous Api
+				genApiProcess.Start ();
+				content = genApiProcess.StandardOutput.ReadToEnd();
+				genApiProcess.WaitForExit ();
+			}
+
+			ValidateIssues(content);
+		}
+
+		// Validates there is no issue or issues found are acceptable
+		void ValidateIssues(string content) {
```

Comment:
Code convention: space before `(`, `{` on a new line *for **methods***.

Reply:
Fixed.